### PR TITLE
dev: move to bash script

### PR DESCRIPTION
**Pull request by tmc**
## Status

Ready for review

## Description of Changes

This unifies the dev/test dependencies for the containerized environments. It also eliminates the dev-shell target in lieu of a bash script which can be used interactively.

## Testing

make test
make dev
./bin/dev-shell ls -al

## Deployment

N/A

## Checklist

N/A

### CODE REVIEW

**Comment by dachary** [comment]
looks great :-) Will take a closer look in the morning.

**Comment by dachary** [comment]
@tmc this is real https://circleci.com/gh/freedomofpress/securedrop/6527?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

<pre>
Error response from daemon: No such container: securedrop-test
docker run \
	-e COMMIT_SHA=0d1f4f022219797bfe5d8ca5718cefeee375434c \
	-e CODECOV_ENV -e CODECOV_TOKEN -e CODECOV_URL -e CODECOV_SLUG -e VCS_COMMIT_ID -e VCS_BRANCH_NAME -e VCS_PULL_REQUEST -e VCS_SLUG -e VCS_TAG -e CI_BUILD_URL -e CI_BUILD_ID -e CI_JOB_ID -e CI -e CIRCLECI -e CIRCLE_BRANCH -e CIRCLE_BUILD_NUM -e CIRCLE_NODE_INDEX -e CIRCLE_PROJECT_USERNAME -e CIRCLE_PROJECT_REPONAME -e CIRCLE_REPOSITORY_URL -e CIRCLE_PR_NUMBER -e CIRCLE_SHA1 -e CIRCLE_ARTIFACTS -e CIRCLE_TEST_REPORTS  \
	--name securedrop-test \
	-ti securedrop-test:0d1f4f022219797bfe5d8ca5718cefeee375434c \
	./bin/test -v tests/functional/test_admin_interface.py tests/functional/test_source_notfound.py tests/functional/test_submission_not_in_memory.py tests/pages-layout/test_journalist.py tests/test_2fa.py tests/test_db.py tests/test_i18n.py tests/test_journalist.py tests/test_manage.py tests/test_securedrop-admin.py tests/test_source.py tests/test_template_filters.py
^@^@./bin/test: line 3: ./dev-deps: No such file or directory
Makefile:23: recipe for target 'test' failed
make: *** [test] Error 1
Exited with code 2
</pre>

File: securedrop/bin/dev-shell [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  args=("bash")
+fi
+docker_args=()
+if [ -z "${SKIP_PORTS:-}" ]; then
+  docker_args+=(
+    -p8080:8080 -p8081:8081 -p5901:5901
+  )
+fi 
+docker_args+=(
+  -v "$(pwd):/app"
```
Comment (dachary) [inline securedrop/bin/dev-shell]: What about -v "$(pwd):$(pwd)" here ? The problem with changing to /app is that the .pyc created in the container won't be useable outside of the container and vice versa.

**Comment by redshiftzero** [comment]
Looks like the holdup here is linting:

```
In ./securedrop/bin/test line 3:

source "${BASH_SOURCE%/*}/dev-deps"

^-- SC1090: Can't follow non-constant source. Use a directive to specify location.
```

**Comment by tmc** [comment]
Whoosps. It's not straightforward to do relative sourcing unfortunately -- maybe we should accept some duplication instead.

**Pull request merged**